code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityExtension.java

```diff
@@ -268,8 +269,10 @@ void handleUpdateIdentities(@NonNull final Event event) {
 			return;
 		}
 
+		// Add pending shared state to avoid race condition between updating and reading identity map
+		final SharedStateResolver resolver = getApi().createPendingXDMSharedState(event);
```

Comment:
would this be fast enough or can we move this before the checks and conversions above as we do in iOS?